cli/flox/src/commands/show.rs

```diff
@@ -148,29 +153,20 @@ fn render_show(search_results: &[SearchResult], all: bool) -> Result<()> {
         .as_ref()
         .map(|d| d.replace('\n', " "))
         .unwrap_or(DEFAULT_DESCRIPTION.into());
-    let versions = if all {
-        let multiple_versions = results
-            .iter()
-            .filter_map(|sr| {
-                let name = sr.rel_path.join(".");
-                // We don't print packages that don't have a version since
-                // the resolver will always rank versioned packages higher.
-                sr.version.clone().map(|version| [name, version].join("@"))
-            })
-            .collect::<Vec<_>>();
-        multiple_versions.join(", ")
-    } else {
-        let sr = results[0];
-        let name = sr.rel_path.join(".");
-        let version = sr.version.clone();
-        if let Some(version) = version {
-            [name, version].join("@")
-        } else {
-            name
-        }
-    };
+    let versions = results
+        .iter()
+        .filter_map(|sr| {
+            let name = sr.rel_path.join(".");
+            // We don't print packages that don't have a version since
+            // the resolver will always rank versioned packages higher.
+            sr.version.clone().map(|version| [name, version].join("@"))
+        })
+        .collect::<Vec<_>>();
+
     println!("{pkg_name} - {description}");
-    println!("    {pkg_name} - {versions}");
+    for version in versions {
```

Comment:
is this non-idiomatic, compared to maniplating the `versions` vec?

Reply:
i think it may just be clearer to do all the processing either a loop or iterator methods. so either:

```rust 
    let versions = results
        .iter()
        .filter_map(|sr| {
            let name = sr.rel_path.join(".");
            // We don't print packages that don't have a version since
            // the resolver will always rank versioned packages higher.
            sr.version.clone().map(|version| [name, version].join("@"))
        })
        .map(|v|, format!("    {v}"))
        .collect::<Vec<_>>();
        
    println!("{pkg_name} - {description}");
    println!("{}" versions.join('\n');

```

or 

```rust
    println!("{pkg_name} - {description}");
    for result in results.iter() {
        let name = result.rel_path.join(".");
        // We don't print packages that don't have a version since
        // the resolver will always rank versioned packages higher.
        let Some(version) = result.version else { continue; }
        println!("    {name}@{version});
    }
```

and in this case i think the latter is a bit more readable but 🤷🏼‍♂️

Reply:
fwiw one could consider the "idiomatic" thing to be a `DisplayResults` struct that implements `Display` which would thus allow testing the of the output in unit tests as a corollary (as would a function that just returns a string but i digress).